forms-flow-web/src/components/ServiceFlow/index.js

```diff
@@ -162,6 +162,16 @@ export default React.memo(() => {
     dispatch(setIsAllTaskVariableExpand(newExpandedState));
   };
 
+  // if all tasks are expanded or collpased individually then update the  allTaskVariablesExpand variable
+  useEffect(() => {
+    if (taskList.every((task) => expandedTasks[task.id])) {
+      dispatch(setIsAllTaskVariableExpand(true));
+    }
+    if (taskList.every((task) => !expandedTasks[task.id])) {
```

Comment:
@shuhaib-aot  already checked that. we need use "if" condition there because if we use "else" case there one of the task is not in expandedTasks[task.id] then setIsAllTaskVariableExpand value will be false and all the task variable will collapse .